coral/src/app/features/connectors/details/ConnectorDetails.tsx

```diff
@@ -41,27 +46,67 @@ function ConnectorDetails(props: ConnectorOverviewProps) {
   const { connectorName } = props;
   const { state: initialEnvironment }: { state: string | null } = useLocation();
 
-  const matches = useMatches();
-  const currentTab = findMatchingTab(matches);
-
+  const [showClaimModal, setShowClaimModal] = useState(false);
   const [environmentId, setEnvironmentId] = useState<string | undefined>(
     initialEnvironment ?? undefined
   );
 
+  const [claimErrorMessage, setClaimErrorMessage] = useState<
+    string | undefined
+  >();
+
+  const toast = useToast();
+
+  const matches = useMatches();
+
+  const currentTab = findMatchingTab(matches);
+
   const {
     data: connectorData,
     isError: connectorIsError,
     error: connectorError,
     isLoading: connectorIsLoading,
     isRefetching: connectorIsRefetching,
-  } = useQuery(["connector-overview", connectorName, environmentId], {
+    refetch: refetchConnectors,
+  } = useQuery({
+    queryKey: ["connector-overview", connectorName, environmentId],
     queryFn: () =>
       getConnectorOverview({
         connectornamesearch: connectorName,
         environmentId,
       }),
   });
 
+  const {
+    mutate: createClaimConnectorRequest,
+    isLoading: isLoadingCreateClaimConnectorRequest,
+    isError: isErrorCreateClaimConnectorRequest,
+  } = useMutation(
+    (remark?: string) =>
+      requestConnectorClaim({
+        connectorName: connectorData?.connectorInfo.connectorName || "",
+        env: connectorData?.connectorInfo.environmentId || "",
+        remark,
+      }),
+    {
+      onSuccess: () => {
+        refetchConnectors().then(() => {
```

Comment:
`refetchConnector` instead of `refetchConnectors` probably?

Reply:
https://github.com/Aiven-Open/klaw/pull/1714/commits/693492b006cdc081a235a32b3cfdcbc92d237c1d